### PR TITLE
[react] Add missing autoComplete and dirName attributes on textarea

### PR DESCRIPTION
**Pull request by devrelm**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
  https://www.w3.org/TR/html5/forms.html#the-textarea-element
  https://html.spec.whatwg.org/multipage/form-elements.html#the-textarea-element

### CODE REVIEW

**Comment by devrelm** [comment]
The failing tests are related to react-router-redux. I'm not sure how my changes could've affected them, but I guess I'll clean those up while I'm at it...

File: types/react/index.d.ts [hunk 1]
```diff
@@ -2815,8 +2816,10 @@ declare namespace React {
     }
 
     interface TextareaHTMLAttributes<T> extends HTMLAttributes<T> {
+        autoComplete?: string;
```
Comment (vsiao) [inline types/react/index.d.ts]: I think this should be `autocomplete`.

**Comment by vsiao** [comment]
can you double check if these attributes are on <input> as well (and if not, add them)? thanks.

**Comment by devrelm** [comment]
@vsiao Yeah, it looks like you're right. I was going off of how it appears elsewhere in the types already, since few browsers actually appear to implement it on textarea elements right now.

I'll update it everywhere and update the pr.

As an aside, the tests are failing due to a change that was made in the redux typings last week (reactjs/redux#2467). It looks like there's an update to the react-router-redux typings that are waiting to be merged (#18071). Once that's done, I'll rebase and push those as well.

**Comment by vsiao** [comment]
great, looks good 👍

**Comment by devrelm** [comment]
This shouldn't have been closed. I mentioned it along with the word "fix" in a comment on a PR in a separate private repo. I guess I have to be more careful what I type out around here...

**Comment by johnnyreilly** [comment]
Thanks!

**Comment by artem-malko** [comment]
@devrelm shouldn't be in lowercase! Actually, you have broken types for react(

**Comment by vsiao** [comment]
That's unfortunate. I would've expected the capitalization to match the DOM API. @eonwhite or @artem-malko please feel free to open a PR to fix.

**Pull request merged**